src/NHibernate/Dialect/SybaseSQLAnywhere17Dialect.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Data;
+using System.Data.Common;
+using NHibernate.Dialect.Schema;
+using Environment = NHibernate.Cfg.Environment;
+
+namespace NHibernate.Dialect
+{
+	/// <remarks>
+	/// The SybaseSQLAnywhere17Dialect uses the SybaseSQLAnywhere12Dialect as its 
+	/// base class. SybaseSQLAnywhere17Dialect includes support for ISO SQL standard
+	/// sequences, which are defined in the catalog table <tt>SYSSEQUENCE</tt>. 
+	/// The dialect uses the SybaseSQLAnywhe12MetaData class for metadata API
+	/// calls, which correctly supports reserved words defined by SQL Anywhere.
+	/// 
+	/// The dialect defaults the following configuration properties:
+	/// <list type="table">
+	///	<listheader>
+	///		<term>Property</term>
+	///		<description>Default Value</description>
+	///	</listheader>
+	///	<item>
+	///		<term>connection.driver_class</term>
+	///		<description><see cref="NHibernate.Driver.SybaseSQLAnywhere17Driver" /></description>
+	///	</item>
+	///	<item>
+	///		<term>prepare_sql</term>
+	///		<description><see langword="false" /></description>
+	///	</item>
+	/// </list>
+	/// </remarks>
+	public class SybaseSQLAnywhere17Dialect : SybaseSQLAnywhere12Dialect
+	{
+		public SybaseSQLAnywhere17Dialect()
+			: base()
+		{
+			DefaultProperties[Environment.ConnectionDriver] = "NHibernate.Driver.SybaseSQLAnywhere17Driver";
+			RegisterDateTimeTypeMappings();
+			RegisterKeywords();
+		}
+
+		new protected void RegisterKeywords()
```

Comment:
`SybaseSQLAnywhere12Dialect` implementation is not what we do usually. Maybe that is the right time to fix it: switch these methods (`RegisterKeywords` and `RegisterDateTimeTypeMappings`) to `override`, calling their base implementation, and remove them from constructor call, both in `SybaseSQLAnywhere12Dialect` and here. `SybaseSQLAnywhere10Dialect` own constructor will call them, once switched to overrides.